### ISSUE TITLE
Do we need to rethink code-alongs

### ISSUE THREAD

**Issue by iteles** [foundersandcoders/master-reference#612]
Opening this issue as I merged in https://github.com/foundersandcoders/master-reference/pull/546 and didn't want this to get lost.

As a Founders & Coders mentor,
I believe the feedback obtained from code-alongs and captured in https://github.com/foundersandcoders/master-reference/pull/546 warrants another look at how we do code-alongs
So that they are an effective learning mechanism for the students.

See: https://github.com/foundersandcoders/master-reference/pull/546/files#diff-ecf680ea6a6dc060749139ff64ac2553R128

**Comment by jsms90** [comment]
Oh yeah! Thanks for raising this!

For full clarity, we are now spending much more time on code alongs than we ever used to. From my observations, I would say this was because FAC10's explanations were far more thorough and in-depth, so that no line of the code was a mystery to students this time.

Whilst code-alongs are now imparting information more successfully than they used to, this pedagogical model is _much_ more mentor-led than peer-led (see #591)

### TL;DR (evidence)

#### Week 2 - testing
@oliverjam created a TDD code along for week 2 (which we didn't have before). This is 2 hours long and I think it may have overrun but I don't quite remember.

#### Week 4 - node 1
As @ZooeyMiller has raised in issue #520, the code along in node week 1 took longer to deliver during this iteration (3.5 hours).

#### Week 6 - postgres
This took 3 hours instead of the [1 hour allotted in the week schedule](https://github.com/foundersandcoders/master-reference/tree/master/coursebook/week-6):
> Alice & Philippa:
> + Code along took a lot longer than planned (10am-1pm). Maybe a lot of time needs to be set aside for that? FAC11 were asking a lot of questions. If you're really going to understand what's going on on every line, you need to ask, but then it inevitably slows things down.

### Week 8 - Express
When we were introduced to hapi.js, we had a one hour code along:
> 10:15 - 11:15 - Hapi Server & Inert code-along

[Express week now has 2 code-alongs](https://github.com/foundersandcoders/master-reference/tree/master/coursebook/week-8), which are 2-3 hours long each:
> 10.30-13:00 - Workshop 1 - Express-Workshop
> 11:00-12:45 - Workshop 3 - Creating and Testing Express Routes

#### Question raised during postgres SGC
Here's the conversation that came about during London's postgres week mentor SGC, which @iteles has linked to above:
> #### Do code alongs even work??
> **Alice:**
> + I find the explanation line by line really really good. But I don't know whether they need to be typing it out themselves at the same time.
>
> **Philippa:**
> + With code along you're concentrating on the syntax without really learning it. Half the time the reason you've missed something is because you've missed it on the screen.
>
> **Jen:**
> + One of the most useful code-along experiences I've had was when other people were having more environment-related problems than me, so I had time to take notes on what the mentor was saying. I have comments for each line of code that was written...until everyone else sorted out their problems and then I was back to frantically writing down the lines as they were speaking.
>
> **Alice:**
> + I would absolutely love it if we had time to take notes.
>
> **Philippa:**
> + Sometimes you don't quite get it because you don't know what's coming up. Like if someone is writing a line where they're initialising a variable and you're going "ok sure...don't know why yet" but you just have to trust that it's necessary & it will all become clear later.  
> + If you were with someone you might say "what they hell is this". But you don't have time and the pressure is on to just keep typing to keep up.
>
> **Philippa:**
> + So maybe if you had the document in front of you, already written, and your job was just to take notes while someone is presenting?
>
> **Jen:**
> + Wouldn't that be even more of a distraction? How can you read through existing code and listen?
>
> **Philippa:**
> + Being able to look through yourself and try to understand what's coming next could feel quite empowering.
>
> **Philippa:**
> + Sometimes you have vague understanding from skimming and then there are things that you need to fill in gaps on. People might have better questions if they've already had a chance to skim it. Going into it blindly, it's just so hectic. So you assume that.
>
> **Jen:**
> + That's really interesting. Sounds a lot like some techniques I was taught on a Teaching English as a Foreign Language course!  
> + And interestingly, that was the exact feedback from a few people in FAC11 about week 3's workshop on HTTP requests & APIs (i.e. readme, then Q&A format). They said that telling people the questions up front let's them know what they're looking for when they do the reading. I think it might even have been the students who have a background in teaching that were making those suggestions.

**Comment by oliverjam** [comment]
@jsms90 My testing codealong didn't quite fit into the allotted two hours. We had to skip the CSS stuff at the end and leave the app unstyled 😢 I think having a screen-recording of myself completing it was very helpful for the cohort as people mentioned going back to watch it when they were confused about things.

**Comment by zurda** [comment]
While doing the testing code along for FAC-16, we also didn't get to the styling. Having said that, I don't think this had a negative affect on the code along. For one thing, I think getting used to not finishing content and having incentive to complete them later on isn't a disadvantage. 
Also, I think that not styling the roman numerals app made sense considering the [week-2 structure](https://github.com/foundersandcoders/master-reference/tree/master/coursebook/week-2). Roman numerals code along is on day 1, and DOM manipulation is on day 2: 

![Screenshot 2019-04-10 at 14 21 17](https://user-images.githubusercontent.com/16784959/55882258-1e20c280-5b9c-11e9-9915-532c315d4d9c.png)

It was really helpful being able to refer students to the youtube video by @oliverjam 👍
